File: tests/java/src/test/java/com/aptos/specs/support/Hooks.java
package com.aptos.specs.support;

import io.cucumber.java.Before;
import io.cucumber.java.After;
import io.cucumber.java.Scenario;

/**
 * Cucumber hooks for setup and teardown.
 */
public class Hooks {
    
    private final World world;
    
    public Hooks(World world) {
        this.world = world;
    }
    
    /**
     * Reset world state before each scenario.
     */
    @Before
    public void beforeScenario(Scenario scenario) {
        world.reset();
    }
    
    /**
     * Cleanup after each scenario.
     */
    @After
    public void afterScenario(Scenario scenario) {
        // Log scenario result if needed
        if (scenario.isFailed() && world.getError() != null) {
            System.err.println("Scenario failed with error: " + world.getError().getMessage());
        }
    }
}

File: tests/kotlin/src/test/kotlin/com/aptos/specs/support/Hooks.kt
package com.aptos.specs.support

import io.cucumber.java.After
import io.cucumber.java.Before
import io.cucumber.java.Scenario

/**
 * Cucumber hooks for scenario lifecycle management.
 */
class Hooks(private val world: World) {
    /**
     * Run before each scenario.
     * Ensures the World is in a clean state.
     */
    @Before
    fun beforeScenario(scenario: Scenario) {
        world.reset()

        // Log scenario start for debugging (optional)
        // println("Starting scenario: ${scenario.name}")
    }

    /**
     * Run after each scenario.
     * Can be used for cleanup or logging.
     */
    @After
    fun afterScenario(scenario: Scenario) {
        // Log scenario result for debugging (optional)
        // println("Finished scenario: ${scenario.name} - ${scenario.status}")

        // Perform any necessary cleanup
        world.reset()
    }
}
